test/k8sT/Policies.go

```diff
@@ -808,3 +815,284 @@ var _ = Describe("K8sValidatedPolicyTestAcrossNamespaces", func() {
+	defer cancel()
+	success, err := k.WaitForPodSuccess(ctx, ns, podName)
+	Expect(err).NotTo(HaveOccurred(), "Pod did not finish as expected.")
+	Expect(success).To(BeFalse(), "Able to connect to service %s on port %d. (It shouldn't)", service.String(), dPort)
+}
+
+func cleanupNetworkPolicy(k *helpers.Kubectl, policy *networkingv1.NetworkPolicy) {
+	By("Cleaning up the policy.")
+	err := k.NetworkingV1().NetworkPolicies(policy.Namespace).Delete(policy.Name, nil)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+var _ = Describe("K8sValidatedPolicyTestNamespaceSelector", func() {
+	var service *v1.Service
+	var podServer *v1.Pod
+	var kub *helpers.Kubectl
+
+	logger := log.WithFields(logrus.Fields{"testName": "K8sValidatedPolicyTestNamespaceSelector"})
+	logger.Info("Starting")
+
+	ns := "namespace-selector-test"
+
+	Context("KubernetesNetworkPolicy between server and client", func() {
+		BeforeEach(func() {
+			kub = helpers.CreateKubectl(helpers.K8s1VMName(), logger)
+
+			By("Creating the namespace that will be used for the pods")
+			_, err := kub.CoreV1().Namespaces().Create(&v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: ns}})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating a simple server that serves on port 80 and 81.")
+			podServer, service = createServerPodAndService(kub, ns, "server", []int{80, 81})
+
+			By("Waiting for pod ready", func() {
+				ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+				defer cancel()
+				err := kub.WaitForPodReady(ctx, ns, podServer.Name)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			By("Testing pods can connect to both ports when no policy is present.")
+			testCanConnect(kub, ns, "client-can-connect-80", service, 80)
+			testCanConnect(kub, ns, "client-can-connect-81", service, 81)
+		})
+
+		AfterEach(func() {
+			cleanupServerPodAndService(kub, podServer, service)
+		})
+
+		It("should support a 'default-deny' policy [Feature:NetworkPolicy]", func() {
+			if helpers.GetCurrentK8SEnv() == "1.6" {
+				Skip("K8s 1.6 doesn't offer a default deny in its policy")
+				//policy := &v1beta1.NetworkPolicy{
+				//	ObjectMeta: metav1.ObjectMeta{
+				//		Name: "allow-ns-b-via-namespace-selector",
+				//	},
+				//	Spec: v1beta1.NetworkPolicySpec{
+				//		// Apply to server
+				//		PodSelector: metav1.LabelSelector{
+				//			MatchLabels: map[string]string{
+				//				"pod-name": podServer.Name,
+				//			},
+				//		},
+				//		// Allow traffic only from NS-B
+				//		Ingress: []v1beta1.NetworkPolicyIngressRule{{
+				//			From: []v1beta1.NetworkPolicyPeer{{
+				//				NamespaceSelector: &metav1.LabelSelector{
+				//					MatchLabels: map[string]string{
+				//						"ns-name": ns,
+				//					},
+				//				},
+				//			}},
+				//		}},
+				//	},
+				//}
+				//policy, err := kub.NetworkingV1().NetworkPolicies(ns).Create(policy)
+				//Expect(err).NotTo(HaveOccurred())
+				//defer cleanupNetworkPolicy(kub, policy)
+			} else {
+				policy := &networkingv1.NetworkPolicy{
```

Comment:
I do not liked this approach, with this code I'm not sure of: 

- What happens if the policy change from versions? As an example, in case of Kuberentes 1.10 introduce a new version, we should use if version in the test? Could we not use the manifest way? 
-  I still do not know why you're not using demo.yaml files, instead of creating two new containers, I think that we should reuse the same manifest in a test.

Reply:
If the kubernetes 1.10 introduces a new version we would write an if statement with that new network policy version.

I think we should diversify the tests and not always use the same `demo.yaml` for all tests